src/routes/index.js

```diff
@@ -1,28 +1,244 @@
+      }
+
+      // TODO: Add unique constraint for username
+
+      const id = uuidv4()
+
+      await User.put({
+        id,
+        username,
+        password,
+      })
+
+      reply.send({
+        token: signToken(id),
+      })
+    })
+  )
+
+  // Sign in
+  fastify.post(
+    '/signin',
+    asyncHandler(async (request, reply) => {
+      const { username, password } = request.body
+
+      if (!username || username === '') {
+        throw new ErrorResponse({
+          title: 'Please provide an username.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'username',
+        })
+      } else if (!password || password === '') {
+        throw new ErrorResponse({
+          title: 'Password is required.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'password',
+        })
+      }
+
+      // TODO: Generate user instead with a random username and password
+      const user = (await User.query(username, { index: 'usersIndex' }))
+        .Items[0]
+
+      if (!user) {
+        throw new ErrorResponse({
+          title: "Username doesn't exist.",
+          code: codes.ERR_NOT_FOUND, //
+          status: statusCodes.NOT_FOUND,
+          pointer: 'username',
+        })
+      }
+
+      const passwordsMatch = Password.compare(user.password, password)
+
+      if (!passwordsMatch) {
+        throw new ErrorResponse({
+          title: 'Username or password is incorrect.',
+          code: codes.ERR_CONFLICT,
+          status: statusCodes.CONFLICT,
+          pointer: 'password',
+        })
+      }
+
+      delete user.password
+
+      reply.send({
+        token: signToken(user.id),
+        data: user,
+      })
+    })
+  )
 
-const routes = {
-  createChat: {
-    method: 'POST',
-    url: '/chat',
-    schema,
-    handler: async (request, reply) => {
-      try {
-        reply.send({ message: 'Chat(WIP)', data })
-      } catch (error) {
-        console.log('error =>', error)
+  // Create chats
+  fastify.post(
+    '/chat',
+    asyncHandler(async (request, reply) => {
+      const { chatName } = request.body
+
+      if (!chatName || chatName === '') {
+        throw new ErrorResponse({
+          title: 'Please provide a chatName.',
+          code: codes.ERR_PARAMETER_REQUIRED,
+          status: statusCodes.UNPROCESSABLE_ENTITY,
+          pointer: 'chatName',
+        })
       }
-    },
-  },
+
+      const chatId = uuidv4()
+
+      // TODO: Add unique constraint for chatName
+      await ChatMessage.put({
+        chatId,
+        chatName,
+      })
+
+      reply.send({
+        data: {
+          chatId,
+          chatName,
+        },
+      })
+    })
+  )
+
+  // Get Messages
+  fastify.get(
+    '/chat/:id/messages',
+    asyncHandler(async (request, reply) => {
+      const chatId = request.params.id
+
+      if (!chatId)
+        connection.socket.send(
```

Comment:
is the error only being sent across the websocket?

Reply:
yes